packages/components/pie-modal/src/index.ts

```diff
@@ -307,26 +301,26 @@ export class PieModal extends RtlMixin(LitElement) implements ModalProps {
     /**
      * Renders the "leadingAction" button if the text is provided.
      *
-     * If `leadingActionText` is not provided, the button is not rendered.
-     * If `leadingActionVariant` is not provided, the default value is used.
-     * The (optional) aria-label is read from the `aria` prop's `leadingActionLabel` property.
+     * If `leadingAction.text` is not provided, the button is not rendered.
+     * If `leadingAction.variant` is not provided, the default value of "primary" is used.
+     * The (optional) aria-label is read from `leadingAction.ariaLabel`.
      *
      * @private
      */
     private renderLeadingAction () : TemplateResult | typeof nothing {
-        if (!this.leadingActionText) {
+        if (!this.leadingAction?.text) {
```

Comment:
Entirely personal preference but I wonder if restructuring the `text`, `variant` and `ariaLabel` properties from the object could be cleaner here. Would make the template render less wordy. Up to you!